README.md

```diff
@@ -69,3 +69,111 @@ versions that you'll need to update.
 * Don't emit unresolved refs as refs to the same pkg
 * If running in Docker, use a m2-srclib directory not inside the repo if in Docker since the Docker source volume is readonly.
 * Simplify the Java 8 JRE bundling process
+
+## Known limitations
+
+### File encoding
+
+Please ensure you have specified file encoding in your .gradle or pom.xml build file, otherwise srclib-java will use platform encoding to compile files and compilation may fail when your files using charset different from the system's one.
+
+### Annotations
+
+We do not support annotations (yet), so if your code relies on annotation processor such as lombok there will be compilation errors and some parts of your code wouldn't be graphed.
+
+### Generated code
+
+If some parts of your code is generated using third party tool or plugin, we may fail to graph your code properly. 
+
+If your project is Maven-based, we are aware about existence of *org.antlr:antlr4-maven-plugin* and will try to generate java files from ANTLR grammar but please make sure that your directory structure matches java package names, i.e. don't put generated classes from foo.bar.baz package to *src/generated*, use *src/generated/foo/bar/baz*. Also please make sure that your ANTLR-based subproject compiles by running `mvn -f SUBPROJECT-DIR/pom.xml generate-sources` command.
+
+If your project is Android-based, we are trying to generate *R.java*, *BuildConfig.java* and aidl-based java files using:
```

Comment:
Android docs use all caps "AIDL" instead of "aidl" and we should be consistent with those.